### PR TITLE
Switch router higher precision

### PR DESCRIPTION
**Pull request by maciek-pioro**
# Description
Has Token Choice router in higher precision.

# Checklist
   - [x] Is this PR focused on a single feature without the possibility of being divided into smaller PRs?
   - [x] Are all variables set with logical defaults that are backward compatible?
   - [x] [I attached link to neptune if it was necessary](https://app.neptune.ai/o/pmtest/org/llm-random/runs/compare?viewId=standard-view&detailsTab=metadata&shortId=LLMRANDOM-5133&dash=charts&type=run&compare=KwRgzADANCM0A&query=(sys%2Fid%3Astring%20%3D%20%22LLMRANDOM-5133%22)%20OR%20(sys%2Fid%3Astring%20%3D%20%22LLMRANDOM-5132%22)%20OR%20(sys%2Fid%3Astring%20%3D%20%22LLMRANDOM-5131%22)%20OR%20(sys%2Fid%3Astring%20%3D%20%22LLMRANDOM-5130%22)&sortBy=%5B%22sys%2Fcreation_time%22%5D&sortFieldType=%5B%22datetime%22%5D&sortFieldAggregationMode=%5B%22auto%22%5D&sortDirection=%5B%22descending%22%5D&suggestionsEnabled=true&lbViewUnpacked=true) 
   - [x] Have you successfully executed the linter and tests?
   - [x] Are all functions concise and don't require splitting?
   - [x] Is there documentation for arguments and complex logic?
   - [x] Are there no temporary solutions in the code? Should a task be added to Trello to refine the code?
   - [x] Are all functions placed in appropriate files?
   - [x] Is the PR name meaningful and descriptive?
   - [x] Is PR description provided, or unnecessary?
  
Other questions:
   - [ ] If I made changes to the requirements or anything related to the Singularity image I did generate new image and upload a new image to the clusters
   - [ ] If this change is significant enough I notified all individuals working with this repository (@channel in slack channel)
   - [ ] If there is a need for a second reviewer I requested additional review 

# Reviewer's checklist:
   - [ ] Does the Neptune link work and does it actually compare the situation before and after the PR?
   - [ ] Do I understand the code?
   - [ ] Do I think another reviewer is not needed or I commented that this PR need another reviewer?

### CODE REVIEW

File: research/conditional/utils/model_utils.py [hunk 1]
```diff
@@ -513,7 +516,7 @@ def get_classes_from_module_names(
     for name in packed_names.split(","):
         if name == "Attention":
             classes.append(llm.Attention)
-        if name == "AttentionMechanism":
+        elif name == "AttentionMechanism":
```
Comment (Simontwice) [inline research/conditional/utils/model_utils.py:519]: omg good catch, what the hell

File: research/conditional/utils/model_utils.py [hunk 2]
```diff
@@ -535,13 +538,21 @@ def get_classes_from_module_names(
...
     return tuple(classes)
 
 
 def get_mixed_precision_ignored_classes(args) -> list[Type[torch.nn.Module]]:
-    ignored_classes = [ExpertGating, LayerNorm, _BatchNorm, torch.nn.Softmax]
+    ignored_classes = [
+        ExpertGating,
+        LayerNorm,
+        _BatchNorm,
+        torch.nn.Softmax,
```
Comment (Simontwice) [inline research/conditional/utils/model_utils.py:553]: I def think this should be reverted (see above also)

File: research/conditional/utils/model_utils.py [hunk 3]
```diff
@@ -535,13 +538,21 @@ def get_classes_from_module_names(
             classes.append(ExpertGating)
         elif name == "Softmax":
             classes.append(torch.nn.Softmax)
```
Comment (Simontwice) [inline research/conditional/utils/model_utils.py:540]: I def think this should be reverted (see below also)

File: research/conditional/utils/model_utils.py (hunk 2 above)
Comment (maciek-pioro) [reply research/conditional/utils/model_utils.py:553]: Aaa nie spushowałem

File: research/conditional/utils/model_utils.py (hunk 3 above)
Comment (maciek-pioro) [reply research/conditional/utils/model_utils.py:540]: Now I don't think so :)

Comment (Simontwice) [reply research/conditional/utils/model_utils.py:540]: sly

**Review by Simontwice — APPROVED** [review]
we gucci

**Pull request merged**